### PR TITLE
minor documentation updates

### PR DESCRIPTION
**Pull request by nixonpjoshua**
these are just some small fixes I forgot to make a PR for I made when I was going to through the SDK documentation

### CODE REVIEW

File: doc/kb_sdk_impl_methods.md [hunk 1]
```diff
@@ -72,8 +72,8 @@ read from the scratch folder.
 
 ```python
 import os
-from AssemblyUtil.AssemblyUtilClient import AssemblyUtil
-from KBaseReport.KBaseReportClient import KBaseReport
+from Assemblyutil.AssemblyutilClient import AssemblyUtil
```
Comment (scanon) [inline doc/kb_sdk_impl_methods.md]: This isn't how it show up for me.  Are you sure?

Comment (scanon) [inline doc/kb_sdk_impl_methods.md]: This doesn't match what I see.  Are you sure?

Comment (scanon) [reply doc/kb_sdk_impl_methods.md]: Same for the next too.

Comment (nlharris) [reply doc/kb_sdk_impl_methods.md]: @nixonpjoshua can you reply to @scanon's questions?

**Comment by nixonpjoshua** [comment]
I think that the issues I had were related to the sdk not caring the case of letters I put in for module names while the filesystem did care once files were created for the modules. This caused some errors for me and I had initially thought that the module names were cased differently as by doing that I was able to make my app work. 

I added a warning into the documentation about this pitfall.

alternatively this could be considered a bug in the SDK and the SDK could be made to only match module names in a case-sensitive way which I believe would make this pitfall no longer exist.

**Comment by nlharris** [comment]
I think @thebenallen made some changes after @nixonpjoshua made a few edits. Can you guys check if I merged the changes correctly?

**Pull request merged**